### DISCUSSION TITLE
[General] <testing upgradeable contract>

### DISCUSSION THREAD

**Discussion by fatemehnedaee** [zkSync-Community-Hub/zksync-developers#137] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I want to write test for my upgradeable contract.

### What is the specific issue or error you're encountering?

When I am getting signers `await ethers.getSigners();` I got this error:
` TypeError: (0 , ethers_1.getAddress) is not a function`

### Can you share the error messages or logs you're receiving, if any?

  "before all" hook in "ZKDinoV1":
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules\.pnpm\@nomicfoundation+hardhat-ethers@3.0.4_ethers@5.7.2_hardhat@2.12.4\node_modules\@nomicfoundation\hardhat-ethers\src\signers.ts:73:30)
      at Function.create (node_modules\.pnpm\@nomicfoundation+hardhat-ethers@3.0.4_ethers@5.7.2_hardhat@2.12.4\node_modules\@nomicfoundation\hardhat-ethers\src\signers.ts:65:12)
      at getSigner (node_modules\.pnpm\@nomicfoundation+hardhat-ethers@3.0.4_ethers@5.7.2_hardhat@2.12.4\node_modules\@nomicfoundation\hardhat-ethers\src\internal\helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules\.pnpm\@nomicfoundation+hardhat-ethers@3.0.4_ethers@5.7.2_hardhat@2.12.4\node_modules\@nomicfoundation\hardhat-ethers\src\internal\helpers.ts:45:30)
      at Context.<anonymous> (test\ZKDinoV1.js:28:45)

### Have you made any recent changes to the contract before encountering this issue?

no

### Are there any external libraries or contracts that your contract interacts with?

no

### Can you provide the relevant portions of your contract code where the issue is occurring?

no

### Have you tried to isolate the problem, and if so, what were the results?

no

### What steps have you already taken to try to resolve the issue?

nothing

**Comment by Romsters** [comment]
Hi @fatemehnedaee, could you please share your code so we can identify and troubleshoot the issue?

**Reply by AlbionaHoti** [reply, marked as answer]
Hi @fatemehnedaee, I found this [tutorial](https://dev.to/abhikbanerjee99/testing-your-upgradeable-smart-contract-2fjd) on dev.to that can be helpful for your case. Please post again if you have any issues building on zkSync Era.

**Comment by andybrierton** [comment]
The error message "TypeError: (0 , ethers_1.getAddress) is not a function" indicates that there is an issue with the getAddress function from ethers.js library. This could be due to a version mismatch or a problem with the installation of ethers.js.

To resolve this issue, make sure you have installed the correct version of ethers.js that is compatible with your Hardhat version. You can try reinstalling ethers.js using the correct version specified in your project dependencies.

Additionally, ensure that there are no conflicts or overrides with other versions of ethers.js in your project dependencies or node_modules folder.

If the issue persists, you may need to check for any custom configurations or modifications that could be causing the problem and revert them to the default settings.

Lastly, consider consulting the documentation or community forums for Hardhat and ethers.js for further assistance on resolving compatibility issues.

**Comment by clevelandmarcus** [comment]
This error indicates that there is an issue with the ethers library in your testing environment. It seems like the getAddress function is not being recognized properly. Try checking your dependencies and ensuring that ethers is installed correctly. Additionally, make sure that you are using compatible versions of ethers and Hardhat.

**Comment by jennifercerso** [comment]
The user is encountering a TypeError: (0 , ethers_1.getAddress) is not a function error when trying to get signers using await ethers.getSigners(); for their upgradeable contract on the testnet environment. The user has not made any recent changes to the contract, and there are no external libraries or contracts that the contract interacts with. The user has not tried to isolate the problem or taken any steps to resolve the issue yet.

**Comment by caojeremyv** [comment]
The user is trying to write a test for their upgradeable contract on a testnet and is encountering an error when getting signers using ethers.getSigners(). The error message states that ethers_1.getAddress is not a function. The user has not made any recent changes to the contract, and there are no external libraries or contracts being used. The user has not provided any relevant code or attempted to isolate the problem.
